src/SuperstateToken.sol

```diff
@@ -583,13 +583,12 @@ contract SuperstateToken is ISuperstateToken, ERC20Upgradeable, PausableUpgradea
             / (usdPerSuperstateTokenChainlinkRaw * stablecoinPrecision);
     }
 
-    /**
-     * @notice The ```subscribe``` function takes in stablecoins and mints SuperstateToken in the proper amount for the msg.sender depending on the current Net Asset Value per Share.
-     * @param inAmount The amount of the stablecoin in
-     * @param stablecoin The address of the stablecoin to calculate with
-     */
-    function subscribe(uint256 inAmount, address stablecoin) external {
-        //@TODO: update function, add support for to address
+    function _subscribe(address to, uint256 inAmount, address stablecoin) internal {
+        if (
```

Comment:
you can save a lot of gas by only doing this check if to != msg.sender

Reply:
so if to != msg.sender && (your current condition)

Reply:
https://github.com/superstateinc/ustb/pull/88/commits/cf04a6f3d1ce98dbce75629ee8752c445f679f5e